src/components/fed/__test__/fed.test.js

```diff
@@ -0,0 +1,141 @@
+import React from "react";
+import { act, fireEvent } from "@testing-library/react";
+import { Fed } from "../";
+import { renderWithProvider, setWindowWidth } from "../../../utils/testing";
+import { getToday, getHour, getUserName } from "../../../utils/fed";
+
+jest.useFakeTimers();
+
+describe("Fed", () => {
+  let wrapper;
+  beforeEach(() => {
+    setWindowWidth(992); //md
+    wrapper = renderWithProvider(<Fed />);
+  });
+
+  it("should show Fed", () => {
+    expect(wrapper).toBeTruthy();
+  });
+
+  describe("show titles and subTitles", () => {
+    it("should show title: Alimentacion", () => {
+      const { getByText } = wrapper;
+      expect(getByText("Alimentacion")).toBeInTheDocument();
+    });
+    it("should show subTitle: AM", () => {
+      const { getByText } = wrapper;
+      expect(getByText("AM")).toBeInTheDocument();
+    });
+    it("should show subTitle: PM", () => {
+      const { getByText } = wrapper;
+      expect(getByText("PM")).toBeInTheDocument();
+    });
+  });
+
+  it("should show CircularProgress", () => {
+    const { container } = wrapper;
+    expect(container.querySelector("circle")).toBeInTheDocument();
+  });
+
+  it("should show Table AM and PM ", () => {
+    const { container } = wrapper;
+    act(() => jest.runAllTimers());
+    // await wait();
+    expect(container.querySelector("table")).toBeInTheDocument();
+  });
+
+  it("should show Switches", () => {
+    const { container } = wrapper;
+    act(() => jest.runAllTimers());
+    expect(container.querySelector("input")).toBeInTheDocument();
+  });
+
+  describe("show days ", () => {
+    it("should show Lunes", () => {
+      const { getAllByText } = wrapper;
+      act(() => jest.runAllTimers());
+      expect(getAllByText("Lunes"));
+    });
+    it("should show Martes", () => {
+      const { getAllByText } = wrapper;
+      act(() => jest.runAllTimers());
+      expect(getAllByText("Martes"));
+    });
+    it("should show Miercoles", () => {
+      const { getAllByText } = wrapper;
+      act(() => jest.runAllTimers());
+      expect(getAllByText("Miercoles"));
+    });
+    it("should show Jueves", () => {
+      const { getAllByText } = wrapper;
+      act(() => jest.runAllTimers());
+      expect(getAllByText("Jueves"));
+    });
+    it("should show Viernes", () => {
+      const { getAllByText } = wrapper;
+      act(() => jest.runAllTimers());
+      expect(getAllByText("Viernes"));
+    });
+    it("should show Sabado", () => {
+      const { getAllByText } = wrapper;
+      act(() => jest.runAllTimers());
+      expect(getAllByText("Sabado"));
+    });
+    it("should show Domingo", () => {
+      const { getAllByText } = wrapper;
+      act(() => jest.runAllTimers());
+      expect(getAllByText("Domingo"));
+    });
+  });
```

Comment:
esto es mejor que quede así si tienes una lista de días en alguna parte, mejor importarlos, supongamos que tienes una tipo 

```js
const days = ['Lunes', 'Martes', ...rest ];
```

```suggestion
  it('renders all week days', () => {
    const { getByText } = wrapper;
    days.forEach(day => {
      expect(getByText(day)).toBeInTheDocument();
    });
  });
```